lincs/src/org/labkey/lincs/view/cromwellSettings.jsp

```diff
@@ -0,0 +1,30 @@
+<%@ page import="org.labkey.api.view.HttpView" %>
+<%@ page import="org.labkey.api.view.JspView" %>
+<%@ page import="org.labkey.lincs.LincsController" %>
+<%@ page extends="org.labkey.api.jsp.FormPage" %>
+<%@ taglib prefix="labkey" uri="http://www.labkey.org/taglib" %>
+<%
+    LincsController.CromwellConfigForm form = ((JspView<LincsController.CromwellConfigForm>) HttpView.currentView()).getModelBean();
+ %>
+<p>
+    Manage Cromwell Settings
+</p>
+<br>
+<labkey:errors/>
+<labkey:form method="post">
+    <table>
+        <tr>
+            <td>Cromwell Server URL:</td>
+            <td><input type="text" name="cromwellServerUrl" value="<%=h(form.getCromwellServerUrl())%>"></td>
+        </tr>
+        <tr>
+            <td>Cromwell Server Port:</td>
+            <td><input type="text" name="cromwellServerPort" value="<%=h(form.getCromwellServerPort())%>"></td>
+        </tr>
+        <tr>
+            <td>Panoama API Key:</td>
+            <td><input type="text" name="apiKey" value="<%=h(form.getApiKey())%>"></td>
```

Comment:
We've been dinged in security reviews for writing previously entered passwords back into configuration forms. We've changed all of our modules' forms. Users have to either resupply the credentials when they save, or there's an explicit checkbox or similar to change the existing credentials. Up to you on how rigorous you want to be here.